plenum/server/node.py

```diff
@@ -1085,9 +1108,19 @@ def msgHasAcceptableViewNo(self, msg, frm) -> bool:
         :return:
         """
         viewNo = getattr(msg, f.VIEW_NO.nm, None)
-        if viewNo is None or not isinstance(viewNo, int) or viewNo < 0:
+        if not self.is_valid_view_or_inst(viewNo):
             return False
-        if viewNo < self.viewNo:
+        # if viewNo < self.viewNo:
+        #     if isinstance(msg, ThreePhaseType):
+        #         inst_id = getattr(msg, f.INST_ID.nm, None)
+        #         if self.is_valid_view_or_inst(inst_id):
+        #             return False
+        #         if len(self.expecting_during_view_change) > inst_id and \
+        #                         self.expecting_during_view_change[inst_id] >= msg.ppSeqNo:
+        #             return True
+        #     self.discard(msg, "un-acceptable viewNo {}"
+        #                  .format(viewNo), logMethod=logger.info)
+        if self.viewNo - viewNo > 1:
```

Comment:
Because even if a node changes view, some requests from previous view might be in flight and since we know that last seq no for any past view, these cannot mislead us. But the node should discard very old messages since it would have moved ahead anyway